components/Modals/BuyingGuideModal.tsx

```diff
@@ -1,27 +1,33 @@
 import { polygon } from "@wagmi/chains";
 import clsx from "clsx";
+import { utils } from "ethers";
 import Image from "next/image";
 import { useContext, useState, useEffect } from "react";
 import { Tooltip } from "react-tooltip";
-import { useAccount, useNetwork, useSwitchNetwork } from "wagmi";
+import { useAccount, useBalance, useNetwork, useSwitchNetwork } from "wagmi";
 
 import { ModalContext } from "@/lib/context";
 import { sliceAddress } from "@/lib/utils";
 import { buyWithUniswap } from "@/payments";
 import { getUSFormattedNumber } from "@/utils";
 
 export default function BuyingGuide() {
-  const { address, connector } = useAccount();
   const { closeModal } = useContext(ModalContext);
   // const formattedGlo = getUSFormattedNumber(glo);
 
+  const { address, connector } = useAccount();
+  const { chain } = useNetwork();
+  const { data: balance } = useBalance({
+    address,
+    token: "0x2791Bca1f2de4661ED88A30C99A7a9449Aa84174", // USDC polygon
```

Comment:
is it the same address for testnet an mainnet?

Reply:
no but I figured the previous dependency is that we request they switch to polygon mainnet right? do we need to support USDC PoS mumbai?

Reply:
Guess it should be fine like that.

Just re-use `USDC_POLYGON_CONTRACT_ADDRESS` as we already have it in utils.